api/specs.proto

```diff
@@ -271,3 +275,11 @@ message ClusterSpec {
 	// TaskDefaults specifies the default values to use for task creation.
 	TaskDefaults task_defaults = 7 [(gogoproto.nullable) = false];
 }
+
+// SecretSpec specifies a user-provided secret.
+message SecretSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	// Secret payload.
+	bytes data = 2;
```

Comment:
nit: naming? Not opposed to `data` but wondering if someone has a better name in mind (@stevvooe ?)

nit: Add a comment to data to explain the maximum allowed length